include/deal.II/matrix_free/constraint_info.h

```diff
@@ -85,15 +85,29 @@ namespace internal
     class ConstraintInfo
     {
     public:
+      /**
+       * Constructor.
+       */
+      ConstraintInfo();
+
+      /**
+       * Set locally owned indices to accelerate the conversion of global to
+       * local indices.
+       */
+      void
+      set_locally_owned_indices(const IndexSet &locally_owned_indices);
+
       /**
        * Version 1: indices are extracted from DoFCellAccessor and
        * constraints are resolved with the help of AffineConstraints.
        */
+      template <typename T = unsigned int>
       void
       reinit(const DoFHandler<dim> &dof_handler,
              const unsigned int     n_cells,
              const bool             use_fast_hanging_node_algorithm = true);
 
+      template <typename T = unsigned int>
       void
       read_dof_indices(
         const unsigned int                                    cell_no,
```

Comment:
I have nothing of consequence to say about the patch itself other than
* You'll want to document this function.
* You'll want to document what `T` stands for (for all of these functions). There may also be a better name than just `T`.
* Giving `T` a default value invites people to omit the argument altogether, and perhaps introduce a bug in the process.